### PR TITLE
[terra-paginator] Update the display order of paginator reduced views

### PR DESCRIPTION
**Pull request by chenyixuan625**
### Summary

**What was changed:**
When paginator rendering as a reduced view, the page label will always show up first and the paginator buttons group together.

**Why it was changed:**
We will want to update the display order of paginator reduced-views, so that the page label will always show up first and the paginator buttons group together(keeping the same order as the default views).

### Testing

For default paginator and the controlled paginator, the previous reduced view:
Screenshot 2023-08-28 at 9 29 02 AM

reduced view after this change:
Screenshot 2023-08-16 at 2 16 01 PM

For progressive paginator and the controlled progressive paginator, the previous reduced view:
Screenshot 2023-08-16 at 2 44 17 PM

reduced view after this change:
Screenshot 2023-08-16 at 2 28 11 PM

Progressive Paginator:

https://github.com/cerner/terra-core/assets/44185348/e6cb3a12-a057-4f46-ae94-dcab927ce515

Controlled Paginator:

https://github.com/cerner/terra-core/assets/44185348/dd69d8e5-8eea-4990-9352-9c1c427cc7e1

This change was tested using:
- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9505 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by chenyixuan625** [comment]
> Do we need to add a left padding before the label in the progressive paginator one? https://raw.githubusercontent.com/cerner/terra-core/0c8e7541596357a05a3f39f29059eef8eedc7601/packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png
> 
> Or, is that the responsibility of the consumer?

When in the reduced views, the default paginator and controlled paginator(label and buttons) are rendered in the center of the paginator previously and I just keep this behavior. For the progressive paginator, to keep consistent with the MPage PaginatorFooter, the label still shows up on the left side and the buttons keep on the right side.

File: packages/terra-paginator/src/ProgressivePaginator.jsx [hunk 1]
```diff
@@ -230,63 +230,69 @@ class ProgressivePaginator extends React.Component {
...
+              disabled={selectedPage === totalPages}
+              onClick={this.handlePageChange(nextPageIndex)}
+              onKeyDown={this.handlePageChange(nextPageIndex)}
+            >
+              <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.paginator.next' })} />
+              <span className={cx('icon')} />
+            </PaginatorButton>
+          </li>
           {
             (totalCount) && (
```
Comment (kolkheang) [inline packages/terra-paginator/src/ProgressivePaginator.jsx:280]: Do we need to have another one bracket and the `totalCount` here? Can we combine with the one above?

File: packages/terra-paginator/src/ProgressivePaginator.jsx [hunk 2]
```diff
@@ -230,63 +230,69 @@ class ProgressivePaginator extends React.Component {
...
-                ariaLabel={intl.formatMessage({ id: 'Terra.paginator.first' })}
-                className={cx(['nav-link', selectedPage === 1 ? 'is-disabled' : null])}
-                disabled={selectedPage === 1}
-                onClick={this.handlePageChange(1)}
-                onKeyDown={this.handlePageChange(1)}
-              >
-                {intl.formatMessage({ id: 'Terra.paginator.first' })}
-              </PaginatorButton>
-            )
+            intl.formatMessage({ id: messageId }, messageAttributes)
```
Comment (kolkheang) [inline packages/terra-paginator/src/ProgressivePaginator.jsx:233]: Do we need to have a left padding style?

File: packages/terra-paginator/src/ProgressivePaginator.jsx (hunk 1 above)
Comment (chenyixuan625) [reply packages/terra-paginator/src/ProgressivePaginator.jsx:280]: This one is for the "Last" button and another one is for the "First" button. We could potentially have something like:
```
const buttonsList = totalCount ?
(
  <FirstButton>
  <PreviousButton>
  <NextButton>
  <LastButton>
) 
: (
  <PreviousButton>
  <NextButton>
)
```
which will duplicate the PreviousButton and NextButton codes, so I just prefer to have both _totalCount_ to make it simple. Thoughts?

File: packages/terra-paginator/src/ProgressivePaginator.jsx (hunk 2 above)
Comment (chenyixuan625) [reply packages/terra-paginator/src/ProgressivePaginator.jsx:233]: replied on the other comment: https://github.com/cerner/terra-core/pull/3897#issuecomment-1697684046

File: packages/terra-paginator/src/ProgressivePaginator.jsx (hunk 1 above)
Comment (kolkheang) [reply packages/terra-paginator/src/ProgressivePaginator.jsx:280]: Oh, I see it now. The `totalCount` value is used to control whether to render the first and last button based on whether the total count is determined or not. Perhaps it's clearer to use a boolean variable name?

```
renderLastButton = totalCount > 0 ? true : false;
```

File: packages/terra-paginator/CHANGELOG.md [hunk 3]
```diff
@@ -4,11 +4,13 @@
 
 * Added
   * Added visually hidden text with meaningful contexts to help screen reader users understand when they navigate the pagination.
+* Changed
```
Comment (vinaybhargavar) [inline packages/terra-paginator/CHANGELOG.md:7]: nit: I am not sure whats being followed in this repo but do we need to add a new line before this?

File: packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png
Comment (vinaybhargavar) [inline packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png:1]: nit: Now the page number seems to be left-aligned. Did we get approval from UI/UX on this?

File: packages/terra-paginator/CHANGELOG.md (hunk 3 above)
Comment (chenyixuan625) [reply packages/terra-paginator/CHANGELOG.md:8]: Fixed: https://github.com/cerner/terra-core/pull/3897/commits/80ad9216dd672e1a59665b19496b871f73b1197a

File: packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png
Comment (chenyixuan625) [reply packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png:1]: For the progressive paginator, we will want to align with the Fusion PaginatorFooter so the page label string is moved to the left side. https://github.com/cerner/terra-core/pull/3897#issuecomment-1697684046

Comment (vinaybhargavar) [reply packages/terra-paginator/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_tiny/paginator-spec/progressive_paginator.png:1]: My bad, apologies for missing this.

**Review by vinaybhargavar — APPROVED** [review]

**Review by eawww — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

File: packages/terra-paginator/translations/de.json [hunk 4]
```diff
@@ -4,6 +4,8 @@
   "Terra.paginator.first": "Erste",
   "Terra.paginator.last": "Letzte",
   "Terra.paginator.pageIndex": "Seite {pageNumber}",
+  "Terra.paginator.pageSelected": "Seite {pageNumber} ist ausgewählt.",
+  "Terra.paginator.pageSelectedWithCount": "Seite {pageNumber} von {pageNumberTotal} ist ausgewählt.",
```
Comment (chenyixuan625) [inline packages/terra-paginator/translations/de.json:8]: Translation files update for UXPLATFORM-9504

**Review by sycombs — APPROVED** [review]

**Pull request merged**